File: PClassic/2022s/Stubs/6/LanternFestival.py
def lanternFestival(n, prefs):
    return -1;

if __name__ == '__main__':
    tests = int(input())
    for test in range(tests):
        n = int(input())
        numEdges = int(input())
        prefs = []
        for e in range(numEdges):
            line = input().split()
            prefs.append([int(line[0]), int(line[1])])
        res = lanternFestival(n, prefs)
        for v in res:
            print(v)

File: PClassic/2022s/Stubs/6/LanternFestival.cpp
#include <iostream>
#include <vector>
#include <list>
#include <algorithm>
#include <cstring>
using namespace std;

vector<int> lanternFestival(int n, vector<vector<int>> prefs)
{
    return null;
}

int main()
{
    int tests;
    cin >> tests;

    while (tests > 0)
    {
        tests--;
        int n;
        cin >> n;
        vector<vector<int>> prefs;
        for(int i = 0; i < n; i++) {
            vector<int> temp;
            prefs.push_back(temp);
        }

        int e;
        cin >> e;
        while (e > 0)
        {
            e--;
            int from, to;
            cin >> from;
            cin >> to;
            prefs[from].push_back(to);
        }
        vector<int> result = lanternFestival(n, prefs);
        for(int i = 0; i < result.size(); i++) {
            cout << result[i] << endl;
        }
        cout << endl;
    }
    return 0;
}